### DISCUSSION TITLE
Extracting raw laptime data

### DISCUSSION THREAD

**Discussion by GeorgeC79** [theOehrly/Fast-F1#142] [Q&A]
Hello. What code can i use to extract the telemetry data from a lap? Not the chart but the numbers that create the chart. Thanks.

**Comment by theOehrly** [comment]
The `Laps` object is a subclass of the `pandas.DataFrame`, which supports exporting to csv, xlsx and other formats. See the documentation of pandas for more information on that.

**Comment by GeorgeC79** [comment]
Thanks for your answer. Can you be more specific or give an example?

**Reply by theOehrly** [reply]
No, can **you** be more specific or give an example of what you want to do? What do you want? Telemetry or lap timing data? How do you want it? What is "raw" data for you?

**Reply by GeorgeC79** [reply]
I want to see/extract the telemetry data from a lap. The raw data that create the plot.

**Comment by markste-in** [comment]
I guess you want the same as in https://github.com/theOehrly/Fast-F1/issues/143#issuecomment-1087246140

If you want to play around with the telemetry data use something like this: **driver_tel will have all the data**
```
event = ff1.get_session(2022,2,"R")
event.load()

for driver in event.laps.Driver.unique():
  for driver_lap in event.laps.pick_driver(driver).iterlaps():
      driver_tel = driver_lap[1].get_car_data()
      # driver tel has your telemetry data .... do something with it here
```

if you just want lap data do this: -> **driver_laps will contain all your data**
```
event = ff1.get_session(2022,2,"R")
event.load()
driver_laps =  event.laps.pick_driver("HAM")
```

**Reply by GeorgeC79** [reply]
Can you give me a complete example of this request? I've been trying for days but it doesn't produce any results. It's blank. Thanks.
